src/updater.c

```diff
@@ -386,101 +141,16 @@ static upd_return_codes_e update_radix_trie (void)
  */
 void *updater_main (void *ptr)
 {
-    char params_splt[LINEMAX];
-    char params_bd[LINEMAX];
-    int num = 0;
-    int update_classifiers = 0;
-    int update_labels = 0;
-    struct configuration *config = ptr;
-
-    /* initialize MD5 digests */
-    memset_s(blacklist_md5, MD5_DIGEST_LENGTH, 0x00, MD5_DIGEST_LENGTH);
-    memset_s(splt_classifier_md5, MD5_DIGEST_LENGTH, 0x00, MD5_DIGEST_LENGTH);
-    memset_s(bd_classifier_md5, MD5_DIGEST_LENGTH, 0x00, MD5_DIGEST_LENGTH);
-
-    /* check for labeling */
-    if (config->label_url) {
-        loginfo("Labels are configured for continuous updating!\n");
-        update_labels = 1;
-    } else {
-        loginfo("Labels are not configured for continuous updating!\n");
-    }
-
-    /* check for remote classifier updates */
-    if (config->params_url && config->params_file) {
-        num = sscanf(config->params_file, "%[^=:]:%[^=:\n#]", params_splt, params_bd);
-        if (num != 2) {
-            loginfo("error: could not parse command \"%s\" into form param_splt:param_bd\n", config->params_file);
-            loginfo("Classifiers are not configured for continuous updating!\n");
-        } else {
-            loginfo("Classifiers are configured for continous updating!\n");
-            update_classifiers = 1;
-        }
-    } else {
-            loginfo("Classifiers are not configured for continous updating!\n");
-    }
-
-    /* initialize the curl library as we need it for downloading updates */
-    if (curl_global_init(CURL_GLOBAL_ALL)) {
-        loginfo("error: curl init failed\n");
-        loginfo("error: updater is not running!\n");
-        return NULL;
-    }
+    struct configuration *config = (struct configuration*)ptr;
```

Comment:
Isn't this configuration_t struct

Reply:
yes, this is valid as well since the typedef is defined as such. I will change to configuration_t though